### ISSUE TITLE
Reduce code footprint when log level is decreased.

### ISSUE THREAD

**Issue by ErezBinyamin** [hideakitai/DebugLog#12]
https://github.com/hideakitai/DebugLog/blob/5bc6d711162fc1f9c3e625b69bf14c1d9283599f/DebugLogEnable.h#L19

## TLDR;
It'd be nice if using less verbose logging actually build a smaller binary.

## Explanation
In the current state if the log level is set to `ERROR` then only `LOG_ERROR` messages will be printed, but all of the other code for the other log messages will still be executed. It just will eventually not print when the [log function](https://github.com/hideakitai/DebugLog/blob/master/DebugLog/Manager.h#L112) is called. Macros like `#ifdef` could be used to **exculde** unused `LOG_XXXX` messages.

The change would be to simply add some code to [DebugLogEnable.h](https://github.com/hideakitai/DebugLog/blob/master/DebugLogEnable.h)
```C
#if defined(DEBUGLOG_DEFAULT_LOG_LEVEL_ERROR)
  #define LOG_WARN(...)
  #define LOG_INFO(...)
  #define LOG_DEBUG(...)
  #define LOG_TRACE(...)
#elif defined(DEBUGLOG_DEFAULT_LOG_LEVEL_WARN)
  #define LOG_INFO(...)
  #define LOG_DEBUG(...)
  #define LOG_TRACE(...)
#elif defined(DEBUGLOG_DEFAULT_LOG_LEVEL_INFO)
  #define LOG_DEBUG(...)
  #define LOG_TRACE(...)
#elif defined(DEBUGLOG_DEFAULT_LOG_LEVEL_DEBUG)
  #define LOG_TRACE(...)
#endif
```

**Comment by hideakitai** [comment]
Looks good. Can I have a pull request?

**Comment by ErezBinyamin** [comment]
Pull request has been made

**Comment by hideakitai** [comment]
Thank you!
